packages/integrations/mdx/README.md

```diff
@@ -78,116 +78,98 @@ Visit the [MDX docs](https://mdxjs.com/docs/what-is-mdx/) to learn about using s
-
-Any additional plugins you apply in your MDX config will be applied *after* your configured Markdown plugins.
-
-#### `astroDefaults`
-
-Astro's MDX files will apply only [Astro's default plugins](/en/reference/configuration-reference/#markdownextenddefaultplugins), without inheriting the rest of your Markdown config. 
-
-This example will apply the default [GitHub-Flavored Markdown](https://github.com/remarkjs/remark-gfm) and [Smartypants](https://github.com/silvenon/remark-smartypants) plugins alongside [`remark-toc`](https://github.com/remarkjs/remark-toc) to your MDX files, while ignoring any `markdown.remarkPlugins` configuration:
-
-```js "extendPlugins: 'astroDefaults'"
+```ts
 // astro.config.mjs
+import { defineConfig } from 'astro/config';
+import mdx from '@astrojs/mdx';
 import remarkToc from 'remark-toc';
+import rehypeMinifyHtml from 'rehype-minify-html';
 
-export default {
-  markdown: {
-    remarkPlugins: [/** ignored */]
-  },
-  integrations: [mdx({
-    remarkPlugins: [remarkToc],
-    // Astro defaults applied
-    extendPlugins: 'astroDefaults',
-  })],
-}
+export default defineConfig({
+  integrations: [
+    mdx({
+      syntaxHighlight: 'shiki',
+      shikiConfig: { theme: 'dracula' },
+      remarkPlugins: [remarkToc],
+      rehypePlugins: [rehypeMinifyHtml],
+      remarkRehype: { footnoteLabel: 'Footnotes' },
+      githubFlavoredMarkdown: false,
+    })
+  ]
+})
 ```
 
-#### `false`
+:::caution
+MDX does not support passing remark and rehype plugins as a "string." You should install, import, and apply the plugin function instead.
+:::
 
-Astro's MDX files will not inherit any [`markdown` options](https://docs.astro.build/en/reference/configuration-reference/#markdown-options), nor will any Astro Markdown defaults be applied:
+📚 See the [Markdown Options reference](https://docs.astro.build/en/reference/configuration-reference/#markdown-options) for a complete list of options.
 
-```js "extendPlugins: false"
-// astro.config.mjs
-import remarkToc from 'remark-toc';
-
-export default {
-  integrations: [mdx({
-    remarkPlugins: [remarkToc],
-    // Astro defaults not applied
-    extendPlugins: false,
-  })],
-}
-```
+### `extendMarkdownConfig`
 
-### `remarkRehype`
+- **Type:** `boolean`
+- **Default:** `true`
 
-Markdown content is transformed into HTML through remark-rehype which has [a number of options](https://github.com/remarkjs/remark-rehype#options).
+MDX will extend [your project's existing Markdown configuration](https://docs.astro.build/en/reference/configuration-reference/#markdown-options) by default. All options will be **deeply merged,** including your remark / rehype plugins and your Shiki configuration.
```

Comment:
```suggestion
MDX will extend [your project's existing Markdown configuration](https://docs.astro.build/en/reference/configuration-reference/#markdown-options) by default. All options will be applied (i.e. **deeply merged**), including your remark / rehype plugins and your Shiki configuration.
```
OK, I'm at a loss here, so maybe you can do better!  :sweat_smile: It would not occur to me that if I set config options, they'd get overwritten. So, it feels very weird to try to say "ALL your preferences will be applied! Honest, we're not gonna overwrite any!" To me, that's a smell that maybe "deeply merged" is one of those "true but not for docs" things? Something that's a little more describing under-the-hood/implementation that might not in fact be necessary to convey how this will affect the reader?

Anyway, I'll leave this for you, but just making you aware that I'm not convinced by this, so I'm calling your attention to it.

Reply:
Deep merge removed! Closing as out-of-date